aspnet/migration/http-modules/sample/Asp.Net5/src/Asp.Net5/Middleware/HttpContextDemoMiddleware.cs

```diff
@@ -0,0 +1,320 @@
+            //<08
+
+            // ----------
+
+            // ASP.NET 4 - HttpContext.Request.UserAgent
+            //>12
+            string userAgent = headersDictionary["User-Agent"].ToString();
+            //<12
+
+            // ASP.NET 4 - HttpContext.Request.UrlReferrer
+            //>13
+            string urlReferrer = headersDictionary["Referer"].ToString();
+            //<13
+
+            // ASP.NET 4 - HttpContext.Request.ContentType 
+            //>14
+            // using Microsoft.Net.Http.Headers;
+
+            MediaTypeHeaderValue mediaHeaderValue = requestHeaders.ContentType;
+            string contentType = mediaHeaderValue?.MediaType;   // ex. application/x-www-form-urlencoded
+            string contentMainType = mediaHeaderValue?.Type;    // ex. application
+            string contentSubType = mediaHeaderValue?.SubType;  // ex. x-www-form-urlencoded
+
+            System.Text.Encoding requestEncoding = mediaHeaderValue?.Encoding;
+            //<14
+
+
+            // ----------
+
+            // ASP.NET 4 - HttpContext.Request.Form 
+            //>15
+            requestHeaders = new RequestHeaders(httpContext.Request.Headers);
+            contentSubType = requestHeaders.ContentType?.SubType;
+
+            if ((contentSubType == "x-www-form-urlencoded") || (contentSubType == "form-data"))
+            {
+                IFormCollection form;
+
+                form = httpContext.Request.Form; // sync
+                // Or
+                form = await httpContext.Request.ReadFormAsync(); // async
+
+                string firstName = form["firstname"];
+                string lastName = form["lastname"];
+            }
+            //<15
+            // ----------
+            // See 
+            // http://stackoverflow.com/questions/31389781/read-body-twice-in-asp-net-5/
+
+            // ASP.NET 4 - HttpContext.Request.InputStream
+            // Unlike reading the form, reading the raw body is not from a buffer.
+            // So you can only do this once per request.
+            // 
+            // If you need to read the body multiple times per request, see
+            // http://stackoverflow.com/questions/31389781/read-body-twice-in-asp-net-5/
+
+            //>16
+            string inputBody;
+            using (var reader = new System.IO.StreamReader(
+                httpContext.Request.Body, System.Text.Encoding.UTF8))
+            {
+                inputBody = reader.ReadToEnd();
+            }
+            //<16
+
+            // Use this middleware as a handler, so no call to 
+            // await _next.Invoke(aspnet5HttpContext);
+
+            // ==================================================================
+            // Response
+
+            // ASP.NET 4 - HttpContext.Response.Status  
+            // ASP.NET 4 - HttpContext.Response.StatusDescription  
+            //>30
+            httpContext.Response.StatusCode = (int)System.Net.HttpStatusCode.OK;
```

Comment:
Replaced HttpStatusCode.Ok with Status200OK.
Used Microsoft.AspNet.... namespace instead of  Microsoft.AspNetCore...., as AspNetCore not yet available as a NuGet package from NuGet.org